### ISSUE TITLE
More active uploads slow down rtorrent?

### ISSUE THREAD

**Issue by xbtowbw** [rakshasa/rtorrent#549]
We have 3 seedboxes all with a 1 Gigabit/s uplink. We try to fully use the UPLINK to seed as much as possible. We can almost fully saturate the links when we have only a few torrents uploading (like 2-3). But the more torrents are actively uploading the less OVERALL bandwidth is used.
1 torrent uploading total 100 MB/s
13 torrents uploading total 35.7 MB/s
17 torrents uploading total 29.6 MB/s
Is that normal? How can we improve this? Systems are quite powerful all with fast raids (raid has lots of bandwidth left, we checked it) and A LOT of ram.

We use Ubuntu 14.04.5 LTS (GNU/Linux 3.16.0-30-generic x86_64) and rtorrent 0.9.6/0.13.6
Our rtorrent.rc looks like this:

# This is an example resource file for rTorrent. Copy to
# ~/.rtorrent.rc and enable/modify the options as needed. Remember to
# uncomment the options you wish to enable.

pieces.preload.type.set = 2
pieces.preload.min_size.set = 262144
pieces.preload.min_rate.set = 5120
network.max_open_files.set = 65000
network.http.max_open.set = 256
network.max_open_sockets.set = 4096
network.receive_buffer.size.set =  128M
network.send_buffer.size.set    = 128M
network.xmlrpc.size_limit.set = 16M

# Maximum and minimum number of peers to connect to per torrent.
min_peers = 500
max_peers = 800

# Same as above but for seeding completed torrents (-1 = same as downloading)
min_peers_seed = 500
max_peers_seed = 800

# Maximum number of simultanious uploads per torrent.
max_uploads = 500

# Global upload and download rate in KiB. "0" for unlimited.
download_rate = 0
upload_rate = 0

# Default directory to save the downloaded torrents.
directory = /media/storage/home/rtorrent/incoming

scgi_port = 127.0.0.1:5000
#scgi_local = /var/lib/rtorrent/rpc.socket
#scgi_local = /tmp/rpc.socket

# Default session directory. Make sure you don't run multiple instance
# of rtorrent using the same session directory. Perhaps using a
# relative path?
session = /media/storage/home/rtorrent/.session

# Watch a directory for new torrents, and stop those that have been
# deleted.
schedule = watch_directory,10,10,load_start=/media/storage/home/rtorrent/watch/*.torrent
#schedule = untied_directory,5,5,stop_untied=

# Close torrents when diskspace is low.
#schedule = low_diskspace,5,60,close_low_diskspace=100M

# The ip address reported to the tracker.
ip = 81.63.186.235
#ip = 212.51.143.250

# The ip address the listening socket and outgoing connections is
# bound to.
#bind = 81.63.186.235
#bind = 212.51.143.250
#bind = rakshasa.no

# Port range to use for listening.
#port_range = 6969-6969
port_range = 6999-6999

# Start opening ports at a random position within the port range.
#port_random = no

# Check hash for finished torrents. Might be usefull until the bug is
# fixed that causes lack of diskspace not to be properly reported.
check_hash = no

# Set whether the client should try to connect to UDP trackers.
use_udp_trackers = yes

# Alternative calls to bind and ip that should handle dynamic ip's.
#schedule = ip_tick,0,1800,ip=rakshasa
#schedule = bind_tick,0,1800,bind=rakshasa

# Encryption options, set to none (default) or any combination of the following:
# allow_incoming, try_outgoing, require, require_RC4, enable_retry, prefer_plaintext
#
# The example value allows incoming encrypted connections, starts unencrypted
# outgoing connections but retries with encryption if they fail, preferring
# plaintext to RC4 encryption after the encrypted handshake
#
encryption = allow_incoming,enable_retry,prefer_plaintext

# Enable DHT support for trackerless torrents or when all trackers are down.
# May be set to "disable" (completely disable DHT), "off" (do not start DHT),
# "auto" (start and stop DHT as needed), or "on" (start DHT immediately).
# The default is "off". For DHT to work, a session directory must be defined.
#
# dht = auto

# UDP port to use for DHT.
#
# dht_port = 6881

# Enable peer exchange (for torrents not marked private)
#
peer_exchange = yes

set_max_file_size = -1
set_split_file_size = -1
log.open_file = "rtorrent", /media/storage/home/rtorrent/rtorrent.log
log.add_output = "debug", "rtorrent"

# max_memory_usage = 104857600

**Comment by chros73** [comment]
> ... all with fast raids (raid has lots of bandwidth left, we checked it) ...

Are you sure about this? Take a look at this issue #443 , and test the disk(s) at first.

**Comment by xbtowbw** [comment]
Yes very sure. 1 Uploading torrent fills the bandwidth (1 Gigabit/s) almost. The more uploading torrents we have the less total bandwidth is used. How can that be???

**Comment by xbtowbw** [comment]
We checked the disk. Its actually a raid and has lots of free resources.
